stubs/setuptools/setuptools/command/upload_docs.pyi

```diff
@@ -8,7 +10,8 @@ class upload_docs(upload):
     user_options: ClassVar[list[tuple[str, str | None, str]]]
     boolean_options: ClassVar[list[str]]
     def has_sphinx(self): ...
-    sub_commands: Any
+    # The callable parameter is self: Self, but using Self still trips up mypy
+    sub_commands: ClassVar[list[tuple[str, Callable[[Self], bool] | None]]]  # type: ignore[misc, assignment]
```

Comment:
I didn't need to modify this file, but I wanted to show what I think may be a mypy issue with `Self` I spotted. pyright looks correct to me.

![image](https://user-images.githubusercontent.com/1350584/221379867-ffec5d01-b714-41c1-b00c-a5e72b3acc32.png)

Example with `sub_commands` removed and inferred from superclass:
![image](https://user-images.githubusercontent.com/1350584/221379941-a42adafa-a67b-4517-99bc-e38e6587dd1e.png)

Reply:
Interesting. I think using `Self` in `ClassVar`s might be underspecified/I'm not sure exactly what the semantics should be really. But this looks fine for now imo.